src/python.cpp

```diff
@@ -942,6 +942,85 @@ SEXP py_to_r(PyObject* x, bool convert) {
   }
 }
 
+// [[Rcpp::export]]
+SEXP py_get_formals(PyObjectRef func, bool convert) {
+  PyObjectPtr inspect(py_import("inspect"));
+  if (inspect.is_null()) stop(py_fetch_error());
+  PyObjectPtr get_signature(PyObject_GetAttrString(inspect.get(), "signature"));
```

Comment:
This is only available since Python 3.3, but I think we need to still support Python 2.7.

Reply:
Python 2’s EOL is in 140 days. Would you like to sign the https://python3statement.org?

Until then we can make this conditional right? Just return NULL if the module isn’t importable?

Reply:
Right, we can just make it conditional.

We'd like to drop Python 2.7 support but as long as some of the larger Python projects like Tensorflow and Keras continue to support Python 2.7 we'll have to support it as well.

Reply:
Actually due to using `tryCatch`, this works now. Every function will simply have

```r
attr(f, 'get_formals_error') <- simpleError("AttributeError: module 'inspect' has no attribute 'signature'")
```